packages/mobile/src/identity/feelessVerification.ts

```diff
@@ -0,0 +1,1061 @@
+  feelessSetCompletedCodes,
+  feelessSetVerificationStatus,
+  feelessUpdateVerificationState,
+  StartVerificationAction,
+  updateE164PhoneNumberSalts,
+} from 'src/identity/actions'
+import { ReactBlsBlindingClient } from 'src/identity/bls-blinding-client'
+import {
+  getAddressesFromLookupResult,
+  lookupAttestationIdentifiers,
+} from 'src/identity/contactMapping'
+import {
+  hasExceededKomenciErrorQuota,
+  KomenciDisabledError,
+  KomenciErrorQuotaExceeded,
+  KomenciSessionInvalidError,
+  PepperNotCachedError,
+  storeTimestampIfKomenciError,
+} from 'src/identity/feelessVerificationErrors'
+import {
+  e164NumberToSaltSelector,
+  E164NumberToSaltType,
+  feelessAcceptedAttestationCodesSelector,
+  feelessAttestationCodesSelector,
+  FeelessVerificationState,
+  feelessVerificationStateSelector,
+  isFeelessVerificationStateExpiredSelector,
+} from 'src/identity/reducer'
+import { startAutoSmsRetrieval } from 'src/identity/smsRetrieval'
+import { VerificationStatus } from 'src/identity/types'
+import {
+  AttestationCode,
+  attestationCodeReceiver,
+  completeAttestations,
+  getActionableAttestations,
+  getAttestationsStatus,
+  NUM_ATTESTATIONS_REQUIRED,
+  requestAndRetrieveAttestations,
+  revealAttestations,
+  VERIFICATION_TIMEOUT,
+} from 'src/identity/verification'
+import { navigate } from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import Logger from 'src/utils/Logger'
+import { setMtwAddress } from 'src/web3/actions'
+import { getContractKit } from 'src/web3/contracts'
+import { registerWalletAndDekViaKomenci } from 'src/web3/dataEncryptionKey'
+import { getConnectedAccount, getConnectedUnlockedAccount } from 'src/web3/saga'
+import { TransactionReceipt } from 'web3-eth'
+
+const TAG = 'identity/feelessVerification'
+
+const KOMENCI_URL = 'https://komenci.celo-networks-dev.org'
+// const KOMENCI_URL = 'http://192.168.86.33:3000'
+// TODO: Populate this with expected implementation address
+const ALLOWED_MTW_IMPLEMENTATIONS: Address[] = ['0x88a2b9B8387A1823D821E406b4e951337fa1D46D']
+const CURRENT_MTW_IMPLEMENTATION_ADDRESS: Address = '0x88a2b9B8387A1823D821E406b4e951337fa1D46D'
+
+export function* feelessFetchVerificationState() {
+  Logger.debug(TAG, '@feelessFetchVerificationState', 'Starting fetch')
+  try {
+    ValoraAnalytics.track(VerificationEvents.verification_fetch_status_start)
+    const [contractKit, walletAddress, feelessVerificationState, e164Number]: [
+      ContractKit,
+      string,
+      FeelessVerificationState,
+      string
+    ] = yield all([
+      call(getContractKit),
+      call(getConnectedUnlockedAccount),
+      select(feelessVerificationStateSelector),
+      select(e164NumberSelector),
+      put(feelessSetVerificationStatus(VerificationStatus.GettingStatus)),
+    ])
+
+    const komenciKit = new KomenciKit(contractKit, walletAddress, {
+      url: KOMENCI_URL,
+      token: feelessVerificationState.komenci.sessionToken,
```

Comment:
We may want to check that the token is set (and is valid).

Reply:
@bowd can correct me if I am wrong but I don't think that's necessary because an empty or expired session token will just be swallowed without error. We make no assumptions about validity and ping Komenci for session status as the source of truth

Reply:
Ok, this makes sense! I see now that it would simply fail a bit further down when it checks the session status 👍